collector/mdadm_linux.go

```diff
@@ -138,12 +163,20 @@ func parseMdstat(mdStatusFilePath string) ([]mdStatus, error) {
 		}
 		currentMD = mainLine[0]               // name of md-device
 		isActive := (mainLine[2] == "active") // activity status of said md-device
+		personality = mainLine[3]             // Personality type of md-device.
```

Comment:
In the check above we only ensure that the line has at least 3 parts. While I assume a line with less than 4 parts to be invalid, it's possible to run into an out-of-bound panic with this change, no?

Bonus points for adding a test for such invalid short lines.